src/group.ts

```diff
@@ -0,0 +1,255 @@
+      }
+      return durationQueries;
+    case 'isnull':
+      typedFilter = filter as IsNullFilter;
+      param = ExpressionSearchMap[`${dataType}.${typedFilter.attribute}`];
+      if (!param) return []; //not a searchable expression
+      return [`${param}:missing=true`];
+    case 'notnull':
+      typedFilter = filter as NotNullFilter;
+      param = ExpressionSearchMap[`${dataType}.${typedFilter.attribute}`];
+      if (!param) return []; //not a searchable expression
+      return [`${param}:missing=false`];
+    case 'equals':
+      typedFilter = filter as EqualsFilter;
+      param = ExpressionSearchMap[`${dataType}.${typedFilter.attribute}`];
+      if (!param) return []; //not a searchable expression
+      return [`${param}=${typedFilter.value}`];
+    case 'value':
+      return valueQueries(filter as ValueFilter, dataType);
+    default:
+      //or, truth, unknown, whatever other type
+      console.warn(`Ignoring "${filter.type}" filter`);
+      return [];
+  }
+}
+
+// Generates an array of fhir query strings for a passed valueFilter
+function valueQueries(filter: ValueFilter, dataType: string): string[] {
+  const param = ExpressionSearchMap[`${dataType}.${filter.attribute}`];
+  if (!param) return []; //not a searchable expression
+
+  // Search prefixes are available for numbers, dates, and quantities https://hl7.org/fhir/R4/search.html#prefix
+  // For boolean and string, only allow equal
+  if (filter.valueBoolean !== undefined) {
+    if (filter.comparator === 'eq') {
+      return [`${param}=${filter.valueBoolean}`];
+    }
+    return [];
+  }
+  if (filter.valueString !== undefined) {
+    if (filter.comparator === 'eq') {
+      return [`${param}=${filter.valueString}`];
+    }
+    return [];
+  }
+  // sa (starts-after) and eb (ends-before) are not used with integer values
+  if (filter.valueInteger !== undefined) {
+    if (filter.comparator === 'sa' || filter.comparator === 'eb') {
+      return [];
+    } else {
+      return [`${param}=${filter.comparator}${filter.valueString}`];
+    }
+  }
+  // quantity [parameter]=[prefix][number]|[system]|[code]
+  if (filter.valueQuantity !== undefined && filter.valueQuantity.value !== undefined) {
+    const systemStr = filter.valueQuantity.system !== undefined ? `|${filter.valueQuantity.system}` : '';
+    const codeStr = filter.valueQuantity.code !== undefined ? `|${filter.valueQuantity.code}` : '';
+    return [`${param}=${filter.comparator}${filter.valueQuantity.value}${systemStr}${codeStr}`];
+  }
+
+  // translate to a quantity to create a query
+  if (
+    filter.valueRatio !== undefined &&
+    filter.valueRatio.numerator?.value !== undefined &&
+    filter.valueRatio.denominator?.value !== undefined
+  ) {
+    // make sure numerator and denominator are comparable
+    const numeratorSystemStr =
+      filter.valueRatio.numerator.system !== undefined ? `|${filter.valueRatio.numerator.system}` : '';
```

Comment:
Do we need `.numerator?.system` like we do above?

Reply:
The `if` statement just above here would not evaluate to true if `filter.valueRatio.numerator` were undefined. In fact, I can probably ditch the first clause of the `if` statement above as well.